ggml.c

```diff
@@ -11656,7 +11656,14 @@ typedef int ggml_lock_t;
 
 #define ggml_lock_init(x)    UNUSED(x)
 #define ggml_lock_destroy(x) UNUSED(x)
+#ifdef __x86_64__
+static inline void ggml_lock_lock(void* x);
+inline void ggml_lock_lock(void* x) {
+    __asm__ __volatile__("pause\n");
```

Comment:
The intrinsic `_mm_pause` should do the same.

Reply:
According to [this](https://stackoverflow.com/questions/7086220/what-does-rep-nop-mean-in-x86-assembly-is-it-the-same-as-the-pause-instru) `rep nop` and `pause` are the same opcode, it is just a quirk of the assembler.

Reply:
Suggestion: change it to use the intrinsic, modify the `#ifdef` to include MSVC. I believe `#if defined(__x86_64__) || defined(_MSC_VER) ` should do it.

Reply:
> Suggestion: change it to use the intrinsic, modify the `#ifdef` to include MSVC. I believe `#if defined(__x86_64__) || defined(_MSC_VER) ` should do it.

Maybe _mm_pause can be used unconditionally if it does not make the code run slower on any architecture?

Reply:
It is still an x86-only intrinsic, it is not available on ARM.

Reply:
> It is still an x86-only intrinsic, it is not available on ARM.

MSVC supports also ARM64, shouldn't there be defined(_M_AMD64)  ?